### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by zhanyuanucb**
Address issue https://github.com/petuum/pirlib/issues/8

### CODE REVIEW

**Comment by hunterhector** [comment]
Suggest to have a public issue for tracking purposes

File: tests/unit/test_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from pirlib.pipeline import pipeline
```
Comment (aurickq) [inline tests/unit/test_pipeline.py:1]: Can we put unit tests in the same folder as the file being tested, using the name "xxx_test.py"

File: tests/unit/test_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    def my_pipeline(out_path: FilePath) -> FilePath:
+        return out_path
+
+    assert my_pipeline.name == "my_pipeline"
+    assert my_pipeline.config.get("key", None) == "value"
+
+def test_pipeline_pkg1():
+    @task
+    def t1(out_path: DirectoryPath) -> DirectoryPath:
+        return out_path
```
Comment (aurickq) [inline tests/unit/test_pipeline.py:20]: Does nesting tasks inside a function actually work?

Comment (aurickq) [reply tests/unit/test_pipeline.py:20]: We need tests that actually run the pipeline too

**Review by aurickq — CHANGES_REQUESTED** [review]

File: tests/unit/test_pipeline.py (hunk 1 above)
Comment (zhanyuanucb) [reply tests/unit/test_pipeline.py:1]: You meant

Comment (zhanyuanucb) [reply tests/unit/test_pipeline.py:1]: You meant something like
```
pirlib/
    pirlib/
    task.py
    task_test.py
    pipeline.py
    pipeline_test.py
```

Comment (aurickq) [reply tests/unit/test_pipeline.py:1]: Right

File: tests/unit/test_pipeline.py (hunk 2 above)
Comment (zhanyuanucb) [reply tests/unit/test_pipeline.py:20]: oops... nested task couldn't get the handler while running

File: tests/unit/test_pipeline.py (hunk 1 above)
Comment (zhanyuanucb) [reply tests/unit/test_pipeline.py:1]: how about the tests for `example/example.py`? Does it also go into `example`?

Comment (zhanyuanucb) [reply tests/unit/test_pipeline.py:1]: any reason why here is better to put unit tests and tested code together?

Comment (aurickq) [reply tests/unit/test_pipeline.py:1]: Mostly down to preference, but this way the unit test file always appears next to the file being tested, and no need to try to maintain a duplicate directory tree under a separate tests folder.

For example.py and other integration tests, can be in a tests folder.

Comment (zhanyuanucb) [reply tests/unit/test_pipeline.py]: updated

**Review by aurickq — APPROVED** [review]

**Pull request merged**